tutorons/python/detect.py

```diff
@@ -0,0 +1,58 @@
+#! /usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+import logging
+import ast
+
+from tutorons.common.extractor import Region
+from tutorons.python.explain import explanations
+
+logging.basicConfig(level=logging.INFO, format="%(message)s")
+
+
+def find_offset(text, call):
+    """Takes in a text and function call and finds the start of call within text"""
+    text_list = text.split('\n')
+    offset = 0
+    for l in range(call.lineno - 1):
+        # add 1 to account for newline characters
+        offset += len(text_list[l]) + 1
+    offset += call.col_offset
+    return offset
+
+
+class BuiltInFinder(ast.NodeVisitor):
+
+    def __init__(self):
+        self.calls = []
+        self.explanations = explanations
```

Comment:
Still uses global data.  Can you just pass in the keys of `explanations` as a list of `builtin_names` to the constructor, from where this is invoked in the PythonBuiltInExtractor?